src/SqlPersistence/ScriptLocation.cs

```diff
@@ -10,30 +9,20 @@ static class ScriptLocation
 
     public static string FindScriptDirectory(ReadOnlySettings settings)
     {
-        var currentDirectory = GetCurrentDirectory(settings);
+        var currentDirectory = GetScriptsRootPath(settings);
         return Path.Combine(currentDirectory, ScriptFolder, settings.GetSqlDialect().Name);
     }
 
-    static string GetCurrentDirectory(ReadOnlySettings settings)
+    static string GetScriptsRootPath(ReadOnlySettings settings)
     {
         if (settings.TryGet("SqlPersistence.ScriptDirectory", out string scriptDirectory))
         {
             return scriptDirectory;
         }
-        var entryAssembly = Assembly.GetEntryAssembly();
-        if (entryAssembly == null)
-        {
-            var baseDir = AppDomain.CurrentDomain.BaseDirectory;
-            var scriptDir = Path.Combine(baseDir, ScriptFolder);
-            //if the app domain base dir contains the scripts folder, return it. Otherwise add "bin" to the base dir so that web apps work correctly
-            if (Directory.Exists(scriptDir))
-            {
-                return baseDir;
-            }
-            return Path.Combine(baseDir, "bin");
-        }
-        var codeBase = entryAssembly.CodeBase;
-        return Directory.GetParent(new Uri(codeBase).LocalPath).FullName;
+
+        //NOTE: This is the same logic that Core uses for finding assembly scanning path.
+        //      RelativeSearchPath is set for ASP .NET and points to the binaries folder i.e. /bin
```

Comment:
Just an FYI, but despite the name, `RelativeSearchPath` contains an absolute path to the bin folder when running in ASP.NET